plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -1,95 +1,82 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 
-# Copyright 2013 Matt Coddington <coddington@gmail.com>
-# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# Copyright 2020 Davinder Pal <dpsangwal@gmail.com>
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-
 DOCUMENTATION = '''
 ---
 module: newrelic_deployment
-author: "Matt Coddington (@mcodd)"
-short_description: Notify newrelic about app deployments
+version_added: "1.2.0"
+author:
+    - "Davinder Pal (@116davinder)"
+    - "Matt Coddington (@mcodd)"
+short_description: Notify newrelic about app deployments using newrelic v2 api
 description:
-   - Notify newrelic about app deployments (see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/deployment-notifications#api)
+   - Notify newrelic about app deployments
+        (https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/record-deployments)
 options:
   token:
-    type: str
     description:
       - API token, to place in the x-api-key header.
     required: true
-  app_name:
     type: str
+  app_name:
     description:
-      - (one of app_name or application_id are required) The value of app_name in the newrelic.yml file used by the application
+      - (one of app_name or application_id is required)
+        The value of app_name in the newrelic.yml file used by the application
     required: false
-  application_id:
     type: str
+  application_id:
     description:
-      - (one of app_name or application_id are required) The application id, found in the URL when viewing the application in RPM
+      - (one of app_name or application_id is required)
+        (see https://rpm.newrelic.com/api/explore/applications/list)
     required: false
-  changelog:
     type: str
+  changelog:
     description:
       - A list of changes for this deployment
     required: false
-  description:
     type: str
+  description:
     description:
       - Text annotation for the deployment - notes for you
     required: false
-  revision:
     type: str
+  revision:
     description:
       - A revision number (e.g., git commit SHA)
-    required: false
-  user:
+    required: true
```

Comment:
This is a breaking change and needs to be documented in the changelog as `breaking_changes`.